observability/runners/data_processor.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+import logging
+import os
+
+from esrally.utils.io import FileSource
+
+logger = logging.getLogger(__name__)
+
+
+class WritableFileSource(FileSource):
+
+    def __init__(self, file_name):
+        super().__init__(file_name, "w")
+
+    def write_lines(self, lines):
+        self.f.writelines([json.dumps(line) + "\n" for line in lines])
+
+    def close(self):
+        self.f.close()
+
+
+class DataProcessor:
+    logger = logging.getLogger(__name__)
+
+    def __init__(self):
+        super().__init__()
+        self._client_index = None
+        self._file_source = None
+        self._output_folder = None
+        self._ops = 0
+
+    def write_meta_file(self):
+        with open(os.path.join(self._output_folder, f"{self._client_index}.metadata"), "w") as meta_file:
+            meta_file.write(str(int(self._ops)))
+
+    async def __call__(self, es, params):
+        es.noop()
```

Comment:
One workaround that I'd try so we can avoid this change in Rally, is:

```python
        rc = es.request_context.get()
        rc["request_start"] = 0
        rc["request_end"] = 0
```

As this is only temporary anyway (can be removed once https://github.com/elastic/rally/issues/335 is implemented) I would try to keep workarounds as contained as possible.